### PR TITLE
[stable/grafana] feat: grafana add sa annotations

### PR DESCRIPTION
**Pull request by Freyert**
#### What this PR does / why we need it:
With the advent of Workload identity it is really important to be able to annotate service accounts. This adds the ability to annotate the Grafana service account.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/grafana/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: grafana
-version: 3.10.2
+version: 3.10.3
```
Comment (zanhsieh) [inline stable/grafana/Chart.yaml]: Need to bump chart version to the latest again.

Comment (Freyert) [reply stable/grafana/Chart.yaml]: @zanhsieh done!

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**